### PR TITLE
CodeGen of With/Lens should work for properties

### PR DESCRIPTION
**Pull request by koffmoff**
As per #622

### CODE REVIEW

File: LanguageExt.CodeGen/CodeGenUtil.cs [hunk 1]
```diff
@@ -38,16 +38,38 @@ public static (ClassDeclarationSyntax PartialClass, TypeSyntax ReturnType, List<
...
+                                         .Where(p => p.AccessorList.Accessors[0].ExpressionBody == null)
+                                         .Where(p => p.AccessorList.Accessors[0].Body == null)
+                                         .Where(p => p.Initializer == null)
+                                         .Select(p => (
+                                             p.Identifier,
+                                             p.Type,
+                                             p.Modifiers
+                                         ));
+ 
+            var fields = variables.Concat(properties).ToList();
```
Comment (louthy) [inline LanguageExt.CodeGen/CodeGenUtil.cs]: This loses the ordering of fields and properties inthe class.  The guidance is that the order the fields are in the class are the order of the entries to the constructor.  Anyone mixing the order of the fields and properties would be confused at this.

One technique might be to zip the members with an `Enumerable.Range(0, Int32.MaxValue)` to create an index for each member and then use that index to sort the final set of results. 

It'd be good to rename `variables`  to fields and `fields` to `members` also.

Comment (koffmoff) [reply LanguageExt.CodeGen/CodeGenUtil.cs]: Done. A bit redundant to impose restrictions on both naming and ordering of the constructor argument, don't you think? But that's out of topic here.

**Comment by louthy** [comment]
Looks good, thanks Fredrik.  I'll try and deploy to nu-get tomorrow. 👍

**Comment by koffmoff** [comment]
Great. Keep up the good work. Much appreciated!

**Pull request merged**